File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> arr) {
    if(arr.size() == 1) arr[0] = -1;
    else
    {
        int mini = arr[0], mI = 0;
        for(int i = 1; i < arr.size(); i++)
        {
            if(mini > arr[i])
            {
                mini = arr[i];
                mI = i;
            }
        }
        arr.erase(arr.begin() + mI);
    }
    return arr;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length -1];
        if(arr.length == 1) {
            answer = new int[1];
            answer[0] = -1;
        } else {
            int min = -1;
        
            int[] backup = new int[arr.length];
            for(int i = 0; i < arr.length; i++) {
               backup[i] = arr[i];
            }
        
            Arrays.sort(arr);
            min = arr[0];
            
            int idx = 0;
            for(int i = 0; i < backup.length; i++) {
                if(backup[i] != min) {
                    answer[idx++] = backup[i];
                }
            }
        }

        return answer;
    }
}